File: mifosng-android/src/main/java/com/mifos/objects/db/Loan.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.objects.db;

import com.google.gson.Gson;

public class Loan {
    public double disbursementAmount;
    public double interestDue;
    public double interestPaid;
    public int loanId;
    public double chargesDue;
    public double totalDue;
    public double principalDue;
    public double principalPaid;
    public String accountId;
    public int accountStatusId;
    public String productShortName;
    public int productId;
    private Currency currency;
    private Client client;
    private String isPaymentChanged;

    public String getIsPaymentChanged() {
        return isPaymentChanged;
    }

    public void setIsPaymentChanged(String isPaymentChanged) {
        this.isPaymentChanged = isPaymentChanged;
    }

    public Client getClient() {
        return client;
    }

    public void setClient(Client client) {
        this.client = client;
    }

    @Override
    public String toString() {
        return new Gson().toJson(this);
    }


    public int getLoanId() {
        return loanId;
    }

    public void setLoanId(int loanId) {
        this.loanId = loanId;
    }

    public Currency getCurrency() {
        return currency;
    }

    public void setCurrency(Currency currency) {
        this.currency = currency;
    }

    public double getDisbursementAmount() {
        return disbursementAmount;
    }

    public void setDisbursementAmount(int disbursementAmount) {
        this.disbursementAmount = disbursementAmount;
    }

    public double getTotalDue() {
        return totalDue;
    }

    public void setTotalDue(double totalDue) {
        this.totalDue = totalDue;
    }

    public int getAccountStatusId() {
        return accountStatusId;
    }

    public String getProductShortName() {
        return productShortName;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Loan loan = (Loan) o;

        if (loanId != loan.loanId) return false;

        return true;
    }

    @Override
    public int hashCode() {
        return loanId;
    }
}

File: core/database/src/main/java/com/mifos/core/objects/db/Loan.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.objects.db

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Loan(
    var disbursementAmount: Double = 0.0,

    var interestDue: Double = 0.0,

    var interestPaid: Double = 0.0,

    var loanId: Int = 0,

    var chargesDue: Double = 0.0,

    var totalDue: Double = 0.0,

    var principalDue: Double = 0.0,

    var principalPaid: Double = 0.0,

    var accountId: String? = null,

    var accountStatusId: Int = 0,

    var productShortName: String? = null,

    var productId: Int = 0,

    var currency: Currency? = null,

    var client: Client? = null,

    var isPaymentChanged: String? = null,
) : Parcelable
